### PR TITLE
Removed hard coded supabase keys

### PR DESCRIPTION
**Pull request by MaxFSP**
## Description ✏️

Closes   Remove hard coded supabase keys in submodule ( issue on trypear/pearai-server ) 

What changed? Feel free to be brief.

Eliminated the need for hard coding of the url and keys, and added function to call them from the server

## Checklist ✅

- [ ] I have added screenshots (if UI changes are present).
- [ ] I have done a self-review of my code.
- [ ] I have manually tested my code (if applicable).

### CODE REVIEW

File: core/db/token.ts [hunk 1]
```diff
@@ -15,37 +12,59 @@ function isTokenExpired(token: string): boolean {
...
-    return Promise.reject('Refresh token is not available');
+    return Promise.reject("Refresh token is not available");
   }
 
   if (isTokenExpired(accessToken)) {
-    console.log('Access token is expired, attempting to refresh');
-    const { data, error } = await supabase.auth.refreshSession({ refresh_token: refreshToken });
+    console.log("Access token is expired, attempting to refresh");
+
+    const supabaseTokens = await requestTokens(accessToken);
```
Comment (nang-dev) [inline core/db/token.ts:44]: okay this shouldnt be in check tokens, the request tokens should be on line 28 i think

and then i dont see create client in the original

**Review by nang-dev — CHANGES_REQUESTED** [review]
we need it to keep lines 4+5

File: core/db/token.ts [hunk 2]
```diff
@@ -15,37 +24,52 @@ function isTokenExpired(token: string): boolean {
...
+  return supabaseTokens;
+}
+
 export async function checkTokens(
   accessToken: string | undefined,
-  refreshToken: string | undefined
-): Promise<{ accessToken: string, refreshToken: string }> {
+  refreshToken: string | undefined,
+): Promise<{ accessToken: string; refreshToken: string }> {
+  await initializeSupabase();
```
[the remark this answers is not in the document]
Comment (MaxFSP) [reply core/db/token.ts:42]: it does run, but if supabase is already initiated then it skips the request

[the remark this answers is not in the document]
Comment (MaxFSP) [reply core/db/token.ts:42]: it basically just acts like another if statement and if its true then it just continues and doesnt request anything to the server

**Comment by nang-dev** [comment]
NICE

**Pull request merged**